changes/21654-install-details

```diff
@@ -0,0 +1,14 @@
+* Ensure details for a software installation run are available and accurate even after the corresponding installer has been edited or deleted
+
+NOTE: The database migration included with this feature does _not_ use activity data to backfill existing installs, due to the lack of indexes on that table.
+
+To backfill install data from activities (which will be more comprehensive and accurate than the migration default, which only populates from installers), run this database query:
+
+```sql
+UPDATE host_software_installs i
+JOIN activities a ON a.activity_type = 'installed_software'
+	AND i.execution_id = a.details->>"$.install_uuid"
+SET i.software_title_name = COALESCE(a.details->>"$.software_title", i.software_title_name),
+	i.installer_filename = COALESCE(a.details->>"$.software_package", i.installer_filename),
+	i.updated_at = i.updated_at
+```
```

Comment:
Maybe document that for existing software installs data will be missing, otherwise customers may have a hard time assessing whether they should do this or not (how broken will the app be if I don't run this...?)

Reply:
It won't be missing. We'll just default to the current value of title/package rather than backfilling what the title/package were when installed (which only affects packages that were re-uploaded as part of an edit).

Reply:
You can see this behavior in the migration test, which includes running this query so we have an automated test for everything we're asking people to run.